tests/upgrade/raycluster_sdk_upgrade_test.py

```diff
@@ -0,0 +1,171 @@
+import requests
+from time import sleep
+
+from torchx.specs.api import AppState, is_terminal
+
+from codeflare_sdk import Cluster, ClusterConfiguration, TokenAuthentication
+from codeflare_sdk.job.jobs import DDPJobDefinition
+
+import pytest
+
+from tests.e2e.support import *
+from codeflare_sdk.cluster.cluster import get_cluster
+
+namespace = "test-ns-rayupgrade"
+
+
+# Creates a Ray cluster
+class TestMNISTRayClusterUp:
+    def setup_method(self):
+        initialize_kubernetes_client(self)
+
+    def test_mnist_ray_cluster_sdk_auth(self):
+        create_namespace_with_name(self, namespace)
+        self.run_mnist_raycluster_sdk_oauth()
+
+    def run_mnist_raycluster_sdk_oauth(self):
+        ray_image = get_ray_image()
+
+        auth = TokenAuthentication(
+            token=run_oc_command(["whoami", "--show-token=true"]),
+            server=run_oc_command(["whoami", "--show-server=true"]),
+            skip_tls=True,
+        )
+        auth.login()
+
+        cluster = Cluster(
+            ClusterConfiguration(
+                name="mnist",
+                namespace=self.namespace,
+                num_workers=1,
+                head_cpus="1",
+                head_memory=2,
+                min_cpus="1",
+                max_cpus=1,
+                min_memory=1,
+                max_memory=2,
+                num_gpus=0,
+                instascale=False,
+                image=ray_image,
+                openshift_oauth=True,
```

Comment:
`openshift_oauth` can be removed here, I believe it will be removed automatically after rebasing.

We are now defaulting to always use openshift_oauth so this option has been removed.
https://github.com/project-codeflare/codeflare-sdk/pull/480